app/data/tasks/fetch_record_csv.py

```diff
@@ -0,0 +1,111 @@
+        'lat': FieldTransform(field='geom', transform=lambda geom: geom.y),
+        'lon': FieldTransform(field='geom', transform=lambda geom: geom.x),
+    }
+
+
+@shared_task
+def export_records(occurred_min, occurred_max, record_label=settings.PRIMARY_RECORD_LABEL):
+    def to_utf8(s):
+        """Convert to utf8 encoding and strip special whitespace/commas for csv writing"""
+        if isinstance(s, str):
+            return re.sub(r'[\r\n\t,]', '', s)
+        elif isinstance(s, unicode):
+            return re.sub(r'[\r\n\t,]', '', s).encode('utf-8')
+        elif s is None:
+            return unicode('').encode('utf-8')
+        else:
+            return re.sub(r'[\r\n\t,]', '', unicode(s)).encode('utf-8')
+
+    records = Record.objects.filter(
+        occurred_from__gte=occurred_min,
+        occurred_to__lte=occurred_max,
+        schema__record_type__label=record_label
+    )
+
+    try:
+        record_type = records[0].schema.record_type
+        schema = record_type.get_current_schema()
+    except IndexError:
+        raise Exception('No records in result set')
+
+    jsonschema = schema.schema
+    details = {
+        key: subschema for key, subschema in jsonschema['definitions'].viewitems()
+        if 'details' in subschema and subschema['details'] is True
+    }
+    details_key = details.keys()[0]
+    record_detail_fields = [
+        key for key, val in details[details_key]['properties'].viewitems()
+        if (
+                'options' not in val or
+                'hidden' not in val['options'] or
+                val['options']['hidden'] is False
+        ) and (
+            key not in DROPPED_KEYS
+        )
+    ]
+    csv_columns = RECORD_FIELDS + record_detail_fields
+    transforms = FIELD_TRANFORMS()
+    row_dicts = []
+    for record in records:
+        row = dict()
+        for field in RECORD_FIELDS:
+            if field in transforms:
+                ft = transforms[field]
+                row[to_utf8(field)] = to_utf8(ft.transform(getattr(record, ft.field)))
+            else:
+                row[to_utf8(field)] = to_utf8(getattr(record, field))
+        for field in record_detail_fields:
+            if field in transforms:
+                ft = transforms[field]
+                if ft.field in record.data['incidentDetails']:
+                    row[to_utf8(field)] = to_utf8(ft.transform(record.data['incidentDetails'][ft.field]))
+            else:
+                if field in record.data['incidentDetails']:
+                    row[to_utf8(field)] = to_utf8(record.data['incidentDetails'][field])
+                else:
+                    row[to_utf8(field)] = ''
+        row_dicts.append(row)
+
+    with tempfile.NamedTemporaryFile() as csvfile:
+        writer = csv.DictWriter(csvfile, fieldnames=csv_columns)
+        writer.writeheader()
+        writer.writerows(row_dicts)
+        store = BlackspotRecordsFile()
+        csvfile.seek(0, 0)
```

Comment:
I'm assuming this is so that the `store.csv.save()` call below can read the whole file. Might be worth a comment, this threw me for a second.